src/br/unicamp/meca/mind/MecaMind.java

```diff
@@ -277,140 +293,189 @@ private void mountMotivationalCodelets() {
 		}
 	}
 
-	private void mountMotivationalBehavioralCodelets() {
-		if (getMotivationalBehavioralCodelets() != null) {
-			for (MotivationalBehavioralCodelet motivationalBehavioralCodelet : getMotivationalBehavioralCodelets()) {
-				if (motivationalBehavioralCodelet != null && motivationalBehavioralCodelet.getId() != null
-						&& motivationalBehavioralCodelet.getMotivationalCodeletsIds() != null
-						&& motivationalBehavioralCodelet.getMotorCodeletId() != null) {
+	private void mountBehaviorCodelets() {
+		if (behaviorCodelets != null) {
+			for (BehaviorCodelet behaviorCodelet : behaviorCodelets) {
+				if (behaviorCodelet != null && behaviorCodelet.getId() != null
+						&& behaviorCodelet.getMotivationalCodeletsIds() != null
+						&& behaviorCodelet.getPerceptualCodeletsIds() != null) {
 
 					/*
 					 * Outputs
-					 */
-					if (motorCodelets != null) {
-						for (MotorCodelet motorCodelet : motorCodelets) {
-							if (motorCodelet != null && motorCodelet.getId() != null) {
-								if (motorCodelet.getId()
-										.equalsIgnoreCase(motivationalBehavioralCodelet.getMotorCodeletId())) {
-									motivationalBehavioralCodelet.addOutputs(motorCodelet.getInputs());
-								}
-							}
-						}
-					}
+					 */										
+					behaviorCodelet.addOutput(actionSequencePlanMemoryContainer);
+					behaviorCodelet.addOutput(actionSequencePlanRequestMemoryContainer);
+					
 					/*
 					 * Inputs
 					 */
-					if (getMotivationalCodelets() != null) {
-						for (MotivationalCodelet motivationalCodelet : getMotivationalCodelets()) {
+					if (motivationalCodelets != null) {
+						for (MotivationalCodelet motivationalCodelet : motivationalCodelets) {
 							if (motivationalCodelet != null && motivationalCodelet.getId() != null) {
-								ArrayList<String> motivationalCodeletsIds = motivationalBehavioralCodelet
+								ArrayList<String> motivationalCodeletsIds = behaviorCodelet
 										.getMotivationalCodeletsIds();
 								if (motivationalCodeletsIds != null) {
 									for (String motivationalCodeletId : motivationalCodeletsIds) {
 										if (motivationalCodeletId != null && motivationalCodelet.getId()
 												.equalsIgnoreCase(motivationalCodeletId)) {
-											motivationalBehavioralCodelet.addInputs(motivationalCodelet.getOutputs());
+											behaviorCodelet.addInputs(motivationalCodelet.getOutputs());
+										}
+									}
+								}
+							}
+						}
+					}
+					
+					if(perceptualCodelets != null) {
+						for(PerceptualCodelet perceptualCodelet : perceptualCodelets) {
```

Comment:
perceptualCodelets.stream().forEach(........

Reply:
Nice suggestion, indeed, but as is just a matter of style, resulting in the same end, will let it to a future PR.